ddtrace/opentelemetry/span_test.go

```diff
@@ -132,6 +134,61 @@ func TestSpanSetName(t *testing.T) {
 	assert.Equal(strings.ToLower("NewName"), p[0]["name"])
 }
 
+func TestSpanLink(t *testing.T) {
+	assert := assert.New(t)
+	t.Setenv("DD_TRACE_DEBUG", "true")
+
+	_, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	_, payloads, cleanup := mockTracerProvider(t)
+	tr := otel.Tracer("")
+	defer cleanup()
+
+	_, sp := tr.Start(context.Background(), "first_span")
+	sp.End()
+	tracer.Flush()
+	_, err := waitForPayload(payloads)
+	assert.NoError(err)
+
+	_, decoratedSpan := tr.Start(context.Background(), "span_with_link",
+		oteltrace.WithLinks(oteltrace.Link{
+			SpanContext: sp.SpanContext(),
+			Attributes:  []attribute.KeyValue{attribute.String("yes", "no")},
+		}))
+
+	decoratedSpan.End()
+	tracer.Flush()
+	payload, err := waitForPayload(payloads)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	assert.NotNil(payload)
+	assert.Len(payload[0][0], 1)
+
+	var spanLinks []ddtrace.SpanLink
+	spanLinkBytes, _ := json.Marshal(payload[0][0]["span_links"])
+	json.Unmarshal(spanLinkBytes, &spanLinks)
+	assert.Len(spanLinks, 1)
+
+	spContext := sp.SpanContext()
+	traceIDbytes := spContext.TraceID()
+	traceIDUpper := binary.BigEndian.Uint64(traceIDbytes[:8])
+	traceIDLower := binary.BigEndian.Uint64(traceIDbytes[8:])
+	assert.Equal(spanLinks[0].TraceID, traceIDLower)
+	assert.Equal(spanLinks[0].TraceIDHigh, traceIDUpper)
+
```

Comment:
For example if this span in [this existing test](https://github.com/DataDog/dd-trace-go/blob/32b50fbc6fde51abf022da5aaabce0d737fba854/ddtrace/opentelemetry/span_test.go#L352) had a span_id of `1 << 64 - 1` the span_id assertion would fail due to precision errors.

Reply:
Discussed this offline. I will set the span_id to a value that be represented in float64 and uint64 without percision errors (ex: span_id = 101)